### PR TITLE
Provide support for External Group operations

### PR DESCRIPTION
**Pull request by mestebangutierrez**
# Description

Provide support for [External Group operations](https://docs.github.com/en/enterprise-cloud@latest/rest/teams/external-groups?apiVersion=2022-11-28#about-external-groups), in particular:
* List external groups in an organization
* Get an external group
* List a connection between an external group and a team
* Connect an external group to a team
* Remove connection between an external group and a team

Fixes https://github.com/hub4j/github-api/issues/1828

# Before submitting a PR:

- [x] Changes must not break binary backwards compatibility. If you are unclear on how to make the change you think is needed while maintaining backward compatibility, [CONTRIBUTING.md](CONTRIBUTING.md) for details.
- [x] Add JavaDocs and other comments explaining the behavior. 
- [x] When adding or updating methods that fetch entities, add `@link` JavaDoc entries to the relevant documentation on https://docs.github.com/en/rest . 
- [x] Add tests that cover any added or changed code. This generally requires capturing snapshot test data. See [CONTRIBUTING.md](CONTRIBUTING.md) for details.
- [x] Run `mvn -D enable-ci clean install site` locally. If this command doesn't succeed, your change will not pass CI.
- [x] Push your changes to a branch other than `main`. You will create your PR from that branch.

# When creating a PR: 

- [x] Fill in the "Description" above with clear summary of the changes. This includes:
  - [x] If this PR fixes one or more issues, include "Fixes #<issue number>" lines for each issue. 
  - [x] Provide links to relevant documentation on https://docs.github.com/en/rest where possible. If not including links, explain why not.
- [x] All lines of new code should be covered by tests as reported by code coverage. Any lines that are not covered must have PR comments explaining why they cannot be covered. For example, "Reaching this particular exception is hard and is not a particular common scenario."
- [x] Enable "Allow edits from maintainers".

### CODE REVIEW

**Comment by mestebangutierrez** [comment]
* Rebased from main
* Added additional tests to improve coverage
* Removed unnecessary code (_e.g._, unrequired bridge methods)
* Minor improvements (make parameters & variables `final` where possible, define `serialVersionUID` for serializable classes)

**Comment by mestebangutierrez** [comment]
@bitwiseman, is anything else required on my side before continuing the contribution process?

**Comment by jbenaventem** [comment]
Please take a moment to review this Pull Request. It contains important changes that could significantly improve the GitHub Enterprise accounts use. If you notice any issues or have any concerns, do not hesitate to comment. Your feedback is greatly appreciated.

File: src/main/java/org/kohsuke/github/GHExternalGroup.java [hunk 1]
```diff
@@ -0,0 +1,263 @@
...
+     * @author Miguel Esteban Gutiérrez
+     */
+    public static class GHLinkedTeam {
+
+        /**
+         * The identifier of the team
+         */
+        @JsonProperty("team_id")
+        @SuppressFBWarnings(value = "UWF_UNWRITTEN_FIELD", justification = "Field comes from JSON deserialization")
+        private long id;
```
Comment (bitwiseman) [inline src/main/java/org/kohsuke/github/GHExternalGroup.java:30]:
```suggestion
        @SuppressFBWarnings(value = "UWF_UNWRITTEN_FIELD", justification = "Field comes from JSON deserialization")
        private long teamId;
```

Comment (bitwiseman) [reply src/main/java/org/kohsuke/github/GHExternalGroup.java:30]: Do this same thing for other fields. Jackson will interpret.  I'm fine with the method names remaining, but would rather not add unneeded annotations.

File: src/main/java/org/kohsuke/github/GHExternalGroups.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * A list of external groups.
+ *
+ * @author Miguel Esteban Gutiérrez
+ */
+public class GHExternalGroups {
```
Comment (bitwiseman) [inline src/main/java/org/kohsuke/github/GHExternalGroups.java:11]: Modify this to match 
https://github.com/hub4j/github-api/blob/main/src/main/java/org/kohsuke/github/GHCommitFilesPage.java

Rename to `GHExternalGroupsPage`
Make `groups` an array.

File: src/main/java/org/kohsuke/github/GHExternalGroupsIterable.java [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+import javax.annotation.Nonnull;
+
+// TODO: Auto-generated Javadoc
+
+/**
+ * Iterable for external group listing.
+ *
+ * @author Miguel Esteban Gutiérrez
+ */
+class GHExternalGroupsIterable extends PagedIterable<GHExternalGroup> {
```
Comment (bitwiseman) [inline src/main/java/org/kohsuke/github/GHExternalGroupsIterable.java:14]:
```suggestion
class GHExternalGroupIterable extends PagedIterable<GHExternalGroup> {
```

File: src/test/java/org/kohsuke/github/EnterpriseManagedSupportTest.java [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+     */
+    @Test
+    public void testIgnoreNonHttpException() throws IOException {
+        GHOrganization org = gitHub.getOrganization(GITHUB_API_TEST_ORG);
+
+        final GHException inputCause = new GHException("Cause");
+        final GHException inputException = new GHException("Test", inputCause);
+
+        final Optional<GHException> maybeException = EnterpriseManagedSupport.forOrganization(org)
+                .handleException(inputException);
```
Comment (bitwiseman) [inline src/test/java/org/kohsuke/github/EnterpriseManagedSupportTest.java:40]: I think `handleException` is more like `filterException`.  Consider rename, but I'm open to leaving if you feel strongly about it.

**Review by bitwiseman — CHANGES_REQUESTED** [review]
Very well tested and implemented. Great job!

A few adjustments and renames needed.

File: src/main/java/org/kohsuke/github/GHExternalGroup.java (hunk 1 above)
Comment (mestebangutierrez) [reply src/main/java/org/kohsuke/github/GHExternalGroup.java]: Done.

File: src/main/java/org/kohsuke/github/GHExternalGroupsIterable.java (hunk 3 above)
Comment (mestebangutierrez) [reply src/main/java/org/kohsuke/github/GHExternalGroupsIterable.java]: Done.

File: src/main/java/org/kohsuke/github/GHExternalGroupsIterable.java [hunk 5]
```diff
@@ -0,0 +1,79 @@
+package org.kohsuke.github;
+
+import java.util.Iterator;
+
+import javax.annotation.Nonnull;
+
+// TODO: Auto-generated Javadoc
```
[the remark this answers is not in the document]
Comment (mestebangutierrez) [reply src/main/java/org/kohsuke/github/GHExternalGroupsIterable.java]: Done.

File: src/test/java/org/kohsuke/github/EnterpriseManagedSupportTest.java (hunk 4 above)
Comment (mestebangutierrez) [reply src/test/java/org/kohsuke/github/EnterpriseManagedSupportTest.java]: Find for me. Renamed.

File: src/main/java/org/kohsuke/github/GHExternalGroups.java (hunk 2 above)
Comment (mestebangutierrez) [reply src/main/java/org/kohsuke/github/GHExternalGroups.java]: Done. I don't like handling arrays, but understand that is a particular pattern used across the client.

**Comment by mestebangutierrez** [comment]
> Very well tested and implemented. Great job!
> 
> A few adjustments and renames needed.

@bitwiseman, first of all, thanks for the review!

I've done all the requested changes, addressing each of them in its own commit.

I've also answered to all the conversations, but haven't resolved them. I let that to you in
case you consider that that changes I've done satisfy the request.

Again, thanks for your effort and patience!

**Comment by mestebangutierrez** [comment]
Rebased again branch from main.

**Comment by mestebangutierrez** [comment]
Rebased from main again.

File: src/main/java/org/kohsuke/github/GHExternalGroups.java (hunk 2 above)
Comment (bitwiseman) [reply src/main/java/org/kohsuke/github/GHExternalGroups.java]: What your change shows is that maybe we don't need to use arrays.  Could you file an issue for this?

File: src/main/java/org/kohsuke/github/GHExternalGroupPage.java [hunk 6]
```diff
@@ -0,0 +1,34 @@
+package org.kohsuke.github;
+
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+
+/**
+ * A list of external groups.
+ *
+ * @author Miguel Esteban Gutiérrez
+ */
+public class GHExternalGroupPage {
```
Comment (bitwiseman) [inline src/main/java/org/kohsuke/github/GHExternalGroupPage.java:10]: I don't think this needs to be public. 
```suggestion
class GHExternalGroupPage {
```

**Review by bitwiseman — APPROVED** [review]

**Pull request merged**